mysite/unmasque/src/core/projection.py

```diff
@@ -107,10 +111,27 @@ def find_projection_dependencies(self, query, s_values):
                 attrib_tup = projection_dep[i][0]
                 projected_attrib.append(attrib_tup[1])
             else:
-                projected_attrib.append('')
+                if len(projection_dep[i]) == 0 and new_result[0][i] != CONST_1_VALUE :
+                    # If no dependency is there and value is not 1 in result this means it is constant.
+                    projected_attrib.append(get_format(projection_coulumn_type[i], new_result[0][i]))
+                else:
+                    # No dependency and value is one so in groupby file will differentiate between count or 1.
+                    projected_attrib.append('')
 
         return projected_attrib, projection_names, projection_dep, True
 
+    def get_projection_column_type(self, query, projection_names):
+        mod_query = query.replace(";",'')
+        datatype_names = []
+        for i in projection_names:
+            typecheck_query = str('SELECT pg_typeof(' + i + ') AS data_type FROM (' + mod_query + ') AS subquery limit 1;')
```

Comment:
later move it to postgres queries class. and create abstract method for it in common queries. These classes should not have anything specific to db engine.